scripts/pylib/pytest-twister-harness/src/twister_harness/device/device_adapter.py

```diff
@@ -98,7 +106,20 @@ def connect(self) -> None:
             msg = 'Cannot connect to not working device'
             logger.error(msg)
             raise TwisterHarnessException(msg)
-        self._connect_device()
+
+        if retry_s > 0:
+            retry_cycles = retry_s * 10
+            for i in range(retry_cycles):
+                try:
+                    self.connect()
```

Comment:
Is this a recursive call?  Should this be self._connect_device() as at line 121?

Reply:
You are correct that this should have been `self._connect_device()`. This was moved from another function into here and I missed updating it. I've also pushed the compliance fix.